Whiskey/Functions/Invoke-WhiskeyTask.ps1

```diff
@@ -178,105 +178,119 @@ function Invoke-WhiskeyTask
         }
     }
 
-    $workingDirectory = $TaskContext.BuildRoot
-    if( $Parameter['WorkingDirectory'] )
-    {
-        $workingDirectory = $Parameter['WorkingDirectory'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'WorkingDirectory'
-    }
-
-    $taskTempDirectory = ''
-    $requiredTools = Get-RequiredTool -CommandName $task.CommandName
-    $startedAt = Get-Date
-    $result = 'FAILED'
+    # Start every task in the BuildRoot.
+    Push-Location $TaskContext.BuildRoot
     $currentDirectory = [IO.Directory]::GetCurrentDirectory()
-    Push-Location -Path $workingDirectory
-    [IO.Directory]::SetCurrentDirectory($workingDirectory)
-    $originalDebugPreference = $DebugPreference
+    [IO.Directory]::SetCurrentDirectory($TaskContext.BuildRoot)
     try
     {
-        if( Test-WhiskeyTaskSkip -Context $TaskContext -Properties $commonProperties)
+        $workingDirectory = $TaskContext.BuildRoot
+        if( $Parameter['WorkingDirectory'] )
         {
-            $result = 'SKIPPED'
-            return
+            # We need a full path because we pass it to `IO.Path.SetCurrentDirectory`.
+            $workingDirectory = 
+                $Parameter['WorkingDirectory'] | 
+                Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'WorkingDirectory' -Mandatory -OnlySinglePath -PathType 'Directory' |
+                Resolve-Path |
+                Select-Object -ExpandProperty 'ProviderPath'
         }
 
-        $inCleanMode = $TaskContext.ShouldClean
-        if( $inCleanMode )
+        $taskTempDirectory = ''
+        $requiredTools = Get-RequiredTool -CommandName $task.CommandName
+        $startedAt = Get-Date
+        $result = 'FAILED'
+        Set-Location -Path $workingDirectory

```

Comment:
`Set/Push/Pop-Location` doesn't also affect WD? :|

Reply:
Thanks, PowerShell! It's awesome, except when it isn't. Then it really isn't.